Source/Base/ASLog.h

```diff
@@ -119,11 +119,44 @@ ASDISPLAYNODE_EXTERN_C_END
  * The logging macros are not guarded by deployment-target checks like the activity macros are, but they are
  * only available on iOS >= 9 at runtime, so just make them conditional.
  */
-#define as_log_create(subsystem, category)  (AS_AT_LEAST_IOS9 ? os_log_create(subsystem, category) : (os_log_t)0)
-#define as_log_debug(log, format, ...)      (AS_AT_LEAST_IOS9 ? os_log_debug(log, format, ##__VA_ARGS__) : (void)0)
-#define as_log_info(log, format, ...)       (AS_AT_LEAST_IOS9 ? os_log_info(log, format, ##__VA_ARGS__) : (void)0)
-#define as_log_error(log, format, ...)      (AS_AT_LEAST_IOS9 ? os_log_error(log, format, ##__VA_ARGS__) : (void)0)
-#define as_log_fault(log, format, ...)      (AS_AT_LEAST_IOS9 ? os_log_fault(log, format, ##__VA_ARGS__) : (void)0)
+
+#define as_log_create(subsystem, category) ({     \
+os_log_t __val;                                   \
```

Comment:
Do the rest of the macros need to pass through their return value as well?

Reply:
If i did not miss anything, other macros don't need to pass through their return value as well.